osu.Game.Rulesets.Osu/Mods/OsuModAutopilot.cs

```diff
@@ -36,21 +39,63 @@ public class OsuModAutopilot : Mod, IApplicableFailOverride, IUpdatableByPlayfie
 
         private int currentFrame;
 
+        int cachedFrames = 1;
+        private const float relax_leniency = 3f;
+
         public void Update(Playfield playfield)
         {
             if (currentFrame == replayFrames.Count - 1) return;
 
-            double time = gameplayClock.CurrentTime;
+            double time = playfield.Clock.CurrentTime;
+
+            bool breakNow = false;
+            bool pause = false;
+            foreach (var h in playfield.HitObjectContainer.AliveObjects.OfType<DrawableOsuHitObject>())
+            {
+                // We are not yet close enough to the object time
+                if(time < h.HitObject.StartTime - relax_leniency) 
+                    break;
+
+                // Already hit or beyond the hittable end time.
+                // Here a relax_leniency is used instead of end time because end time causes lags for some reason
```

Comment:
what does "for some reason" mean?

Reply:
When i used the end time as a check, as done in osu relax, the cursor will lag way behind the position it is supposed to be. I'm not really sure why it lagged behind so much. The relax_leniency is intended to be used as an offset to the start time, where the click may register as great or meh still.